File: 단계별로 풀어보기/정렬/10814/나이순 정렬.cpp
#include <iostream>
#include <list>
#include <string>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    int age;
    string name;
    list<string> ages[200];
    list<string>::iterator it;
    cin >> N;
    
    for(int i = 0; i < N; i++){
        cin >> age >> name;
        ages[age-1].push_back(name);
    }
    
    for(int i=0; i < 200; i++){
        for(it = ages[i].begin(); it != ages[i].end(); it++){
            cout << i+1 << ' '<< *it <<'\n';
        }
    }
    
    return 0;
}

File: 단계별로 풀어보기/정렬/10814/나이순 정렬.py
import sys
    
def init():
    N = int(sys.stdin.readline())
    ages = [list() for i in range(200)]
    
    for i in range(N):
            age, name = sys.stdin.readline().split()
            age = int(age)
            ages[age-1].append(name)

    for i in range(200):
        for j in range(len(ages[i])):
            sys.stdout.write('%d %s\n' %(i+1,ages[i][j]))
                
init()
